src/models/token.rs

```diff
@@ -29,41 +26,35 @@ pub struct ApiToken {
     pub revoked: bool,
 }
 
-diesel::sql_function! {
-    fn digest(input: Text, method: Text) -> Bytea;
-}
-
 impl ApiToken {
     /// Generates a new named API token for a user
     pub fn insert(conn: &PgConnection, user_id: i32, name: &str) -> AppResult<CreatedApiToken> {
-        let plaintext = format!(
-            "{}{}",
-            TOKEN_PREFIX,
-            crate::util::generate_secure_alphanumeric_string(TOKEN_LENGTH)
-        );
+        let token = SecureToken::generate(SecureTokenKind::API);
 
         let model: ApiToken = diesel::insert_into(api_tokens::table)
             .values((
                 api_tokens::user_id.eq(user_id),
                 api_tokens::name.eq(name),
-                api_tokens::token.eq(digest(&plaintext, "sha256")),
+                api_tokens::token.eq(token.sha256()),
             ))
             .get_result(conn)?;
 
-        Ok(CreatedApiToken { plaintext, model })
+        Ok(CreatedApiToken {
+            plaintext: token.plaintext().into(),
```

Comment:
not terribly important but I'm wondering if it would be possible to avoid the string clone here. if I understand correctly the `token` is not used beyond this point so it should be possible to consume it to get the original plaintext `String`, right?

Reply:
Yes, but we would need to move it out of the token. I also don't think this is really important.